tests/src/test/java/alluxio/client/cli/JournalToolTest.java

```diff
@@ -254,10 +256,27 @@ private long getCurrentCopyCatSnapshotIndex(String journalFolder) throws Throwab
     return currentSnapshotIdx.get();
   }
 
-  private String findCheckpointDir() throws IOException {
-    List<Path> checkpoint = Files.list(mDumpDir.toPath())
-        .filter(p -> p.toString().contains("checkpoints-")).collect(toList());
-    assertEquals("Unexpected checkpoint list: " + checkpoint, 1, checkpoint.size());
+  private String findCheckpointDir() throws Exception {
+    final List<Path> checkpoint = new ArrayList<>();
+    CommonUtils.waitFor("Checkpoint list:", () -> {
```

Comment:
Can we just do the "wait" part for getting the list of paths? Once we get the non-empty list of paths, we can just do the previous/existing assert of the size? The current implementation of this function is doing both the condition to terminate the wait, as well as asserts. I'd prefer this wait function only determines the termination condition for the wait, and do the assert outside of the wait.